src/main/java/com/indeed/jiraactions/api/response/issue/Issue.java

```diff
@@ -19,20 +28,61 @@
     public Field fields;
     public ChangeLog changelog;
 
+    // Multivalued rich object fields are treated different in Jira changelog histories than
+    //  multivalued primitive fields.
+    private static Set<String> MULTIVALUED_RICH_FIELDS = ImmutableSet.of(
+            "component"
+    );
+
     public String initialValue(final String field) throws IOException {
         return initialValue(field, field);
     }
 
+    /**
+     * Determines the initial value of the field based on current issue state and recorded changelog.
+     */
     public String initialValue(final String field, final String fallbackField) throws IOException {
-        final Item history = initialItem(false, field);
-        if(history != null) {
-            return history.fromString == null ? "" : history.fromString;
+        if (MULTIVALUED_RICH_FIELDS.contains(field)) {
+            // Multivalued rich fields' initial state must be determined by inference from
+            //  taking the current state and walking the history to reverse the actions.
+            List<String> values =
+                    null == this.fields.components ?
+                            Lists.newArrayList() :
+                            ImmutableList.copyOf(this.fields.components).stream()
+                                    .map(component -> component.name)
+                                    .collect(toList());
+
+            if (null != this.changelog && null != this.changelog.histories) {
+                for (int i = this.changelog.histories.length - 1; i >= 0; i--) {
+                    final History history = this.changelog.histories[i];
+                    for (final Item item: history.getAllItems("component")) {
+                        // Reverse the action.
+                        if (null == item.to) {
```

Comment:
I think this is supposed to be `item.toString` not `item.to`. (`item.to` will have the identifier; `item.toString` has the actual name of the component.)